src/pmpo_c.cpp

```diff
@@ -374,6 +424,51 @@ void polympo_getMPVel_f(MPMesh_ptr p_mpmesh, const int nComps, const int numMPs,
   p_MPs->parallel_for(getMPVel, "getMPVel");
   kkDbl2dViewHostU arrayHost(mpVelHost,nComps,numMPs);
   Kokkos::deep_copy(arrayHost, mpVelCopy);
+  pumipic::RecordTime("PolyMPO_getMPVel", timer.seconds());
+}
+
+//TODO: implement these
+void polympo_setMPStrainRate_f(MPMesh_ptr p_mpmesh, const int nComps, const int numMPs, const double* mpStrainRateIn){
+  checkMPMeshValid(p_mpmesh);
+  auto p_MPs = ((polyMPO::MPMesh*)p_mpmesh)->p_MPs;
+  PMT_ALWAYS_ASSERT(nComps == 6); //TODO: mp_sym_mat3d_t
+  PMT_ALWAYS_ASSERT(numMPs >= p_MPs->getCount());
+  PMT_ALWAYS_ASSERT(numMPs >= p_MPs->getMaxAppID());
+
+  auto mpStrainRate = p_MPs->getData<polyMPO::MPF_Vel>();
+  auto mpAppID = p_MPs->getData<polyMPO::MPF_MP_APP_ID>();
+  kkViewHostU<const double**> mpStrainRateIn_h(mpStrainRateIn,nComps,numMPs);
+  Kokkos::View<double**> mpStrainRateIn_d("mpStrainRateDevice",nComps,numMPs);
+  Kokkos::deep_copy(mpStrainRateIn_d, mpStrainRateIn_h);
+  auto setMPStrainRate = PS_LAMBDA(const int& elm, const int& mp, const int& mask){
+    if(mask){
+      mpStrainRate(mp,0) = mpStrainRateIn_d(0, mpAppID(mp));
+      mpStrainRate(mp,1) = mpStrainRateIn_d(1, mpAppID(mp));
+      mpStrainRate(mp,2) = mpStrainRateIn_d(2, mpAppID(mp));
+      mpStrainRate(mp,3) = mpStrainRateIn_d(3, mpAppID(mp));
+      mpStrainRate(mp,4) = mpStrainRateIn_d(4, mpAppID(mp));
+      mpStrainRate(mp,5) = mpStrainRateIn_d(5, mpAppID(mp));
+    }
+  };
+  p_MPs->parallel_for(setMPStrainRate, "setMPStrainRate");
+}
+void polympo_getMPStrainRate_f(MPMesh_ptr p_mpmesh, const int nComps, const int numMPs, double* mpStrainRateHost){
+  (void)p_mpmesh;
+  (void)nComps;
+  (void)numMPs;
+  (void)mpStrainRateHost;
+}
+void polympo_setMPStress_f(MPMesh_ptr p_mpmesh, const int nComps, const int numMPs, const double* mpStressIn){
+  (void)p_mpmesh;
+  (void)nComps;
+  (void)numMPs;
+  (void)mpStressIn;
+}
+void polympo_getMPStress_f(MPMesh_ptr p_mpmesh, const int nComps, const int numMPs, double* mpStressHost){
```

Comment:
We need a call to checkMPMeshValid(...). And throw a message this routine is not implemented and exit code.